### PR TITLE
feat: coupon improvement 2.16.1

### PR DESCRIPTION
**Pull request by virginiaritonga**
feat(coupon): Implement coupon improvement v2.16.x

Summary:
improvement new coupon popup all page

changes:
- improvement changes 2.16.x
- with breaking changes at [notion](https://www.notion.so/sirclo/v2-16-0-5bc6e851b5834083950df5a8dbe2f9a3)

User Acceptance Criteria :
- [x] Feature popup coupon : https://phabricator.sirclo.com/T34287
- [x]  Feature popup detail coupon : https://phabricator.sirclo.com/T34285
- [x]  Feature coupon secara default bisa di apply : https://phabricator.sirclo.com/T31416
- [x]  UI nya di [figma](https://www.notion.so/sirclo/Sentinel-Coupon-Improvement-All-Templates-7a805ba955b04a2f9ccdcfa06cb5181b) ya
- [x]  Figma Rose https://www.figma.com/file/cl2GPB5tKKSNjA5CxPsJBB/%5BVENDOR%5D-Ros%C3%A9?node-id=3103%3A4400

**Props**
--
1. Improvement loading component coupon
2. Add onSuccessCopyCodeCoupon
3. Remove isCouponAccordion
4. Remove withCouponTitle
5. Add icon props copyIcon

**Location changes**
--
1. OrderSummary
2. ListPaymentMethod

**ScreenShoot**
--
  - PopUp Coupons Full Screen (Dekstop)
Screen Shot 2022-10-25 at 16 45 25
Screen Shot 2022-10-25 at 16 45 32
Screen Shot 2022-10-25 at 16 59 07
Screen Shot 2022-10-25 at 17 23 59

  - PopUp Coupons Full Screen (Mobile)
Tangkapan Layar 2022-10-25 pada 16 45 48
Tangkapan Layar 2022-10-25 pada 16 59 00

  - PopUp Detail Coupons Full Screen (Dekstop)

Screen Shot 2022-10-25 at 16 46 07

  - PopUp Detail Coupons Full Screen (Mobile)
Tangkapan Layar 2022-10-25 pada 16 46 16

Additional Test:
Points
Tangkapan Layar 2022-10-25 pada 16 48 35
Screen Shot 2022-10-25 at 17 24 15

### CODE REVIEW

File: public/scss/components/OrderSummaryBox.module.scss [hunk 1]
```diff
@@ -425,6 +414,46 @@
...
+        @include flex(column, center, center);
+        position: relative;
+        width: 100%;
+        padding: calc(8px + 0.8vh) calc(8px + 0.8vw) calc(26px + 2.6vh);
+
+        & > * 
+        {
+        width: 100%;
+        max-width: 1170px; 
+        margin: 0;
```
Comment (falmesino) [inline public/scss/components/OrderSummaryBox.module.scss:429]:
```suggestion
            width: 100%;
            max-width: 1170px; 
            margin: 0;

```

indent-nya kurang nih nyonya

File: public/scss/components/OrderSummaryBox.module.scss [hunk 2]
```diff
@@ -796,6 +874,152 @@
...
+    {
+        @include flex(column, flex-start, flex-start);
+        @include fixed(auto, auto, 0);
+        z-index: 1;
+        margin: 0;
+        background-color: $color_white;
+        border-radius: 2px;
+
+        max-width: 100vw;
+        max-height: 80vh;;
```
Comment (falmesino) [inline public/scss/components/OrderSummaryBox.module.scss:927]:
```suggestion
        max-height: 80vh;

```

titik koma-nya kebanyakan

File: public/scss/components/OrderSummaryBox.module.scss [hunk 3]
```diff
@@ -796,6 +874,152 @@
...
+            }
+    }
+
+    &_voucherDetailPopUpCodeTitle
+    {
+        @include typographyBuilder(secondary,400,12,16);
```
Comment (falmesino) [inline public/scss/components/OrderSummaryBox.module.scss:982]:
```suggestion
        @include typographyBuilder(secondary, 400, 12, 16);

```

kasih spasi

File: public/scss/components/OrderSummaryBox.module.scss [hunk 4]
```diff
@@ -796,6 +874,152 @@
...
+        margin-bottom: 0;
+        word-break: break-all;
+        color: $color_text_primary;
+    }
+
+    &_voucherDetailPopUpClose,
+    &_voucherDetailPopUpCodeCopyButton
+    {
+  
+        border: none;
```
Comment (falmesino) [inline public/scss/components/OrderSummaryBox.module.scss:1008]:
```suggestion
        border: none;

```

hapus enternya

**Review by falmesino — CHANGES_REQUESTED** [review]

**Comment by sultanfariz** [comment]
terms and condition kupon kelebihan padding sama list stylenya ngga decimal
image

**Comment by mayun19** [comment]
warna button disable "gunakan" berbeda dengan yg ada di figma 
> Screen Shot 2022-10-25 at 16 45 25
> 

**figma** 
image

File: components/OrderSummaryBox/index.tsx [hunk 5]
```diff
@@ -194,7 +222,7 @@ const OrderSummaryComponent: FC<iProps> = ({
               ...classesOrderSummary,
               voucherTextClassName: `${styleOrderSummary.orderSummary_voucherText} ${lng}`,
               voucherAppliedTextClassName: `${styleOrderSummary.orderSummary_voucherAppliedText} ${lng}`,
-              voucherTitleClassName: `${styleOrderSummary.orderSummary_voucherTitle} ${lng}`,
+              // voucherTitleClassName: `${styleOrderSummary.orderSummary_voucherTitle} ${lng}`,
```
Comment (mayun19) [inline components/OrderSummaryBox/index.tsx:225]: ini lupa dihapus kah?

File: public/scss/components/OrderSummaryBox.module.scss [hunk 6]
```diff
@@ -796,6 +874,152 @@
...
+        @include flex(row, center, center);
+        margin: 8px 0 calc(26px + 2.6vh);
+    }
+
+    &_voucherShowMoreButton
+    {
+        @include typographyBuilder(secondary, 400, 12, 16);
+        color: $color_rose;
+        padding: 0;
+        outline: 0;
```
Comment (mayun19) [inline public/scss/components/OrderSummaryBox.module.scss:888]: bisa diseragamkan dg outline: none ?

File: public/scss/components/OrderSummaryBox.module.scss [hunk 7]
```diff
@@ -796,6 +874,152 @@
...
+        border: 0;
```
Comment (mayun19) [inline public/scss/components/OrderSummaryBox.module.scss:889]: bisa diseragamkan dg border: none ?

File: public/scss/components/OrderSummaryBox.module.scss [hunk 8]
```diff
@@ -796,6 +874,152 @@
...
+
+    &_voucherDetailPopUpHeaderTitle
+    {
+        @include typographyBuilder(secondary, 500, 14, 24);
+        margin: 0;
+    }
+
+    &_voucherDetailPopUpDescContainer
+    {
+        @include typographyBuilder(secondary,400,14,24);
```
Comment (mayun19) [inline public/scss/components/OrderSummaryBox.module.scss:953]: ini dikasih spasi juga

@include typographyBuilder(secondary, 400, 14, 24);

**Review by mayun19 — CHANGES_REQUESTED** [review]

**Comment by virginiaritonga** [comment]
updated styling:

- disabled button

  Screen Shot 2022-10-26 at 10 19 22

- terms condition list
  Screen Shot 2022-10-26 at 09 46 00

**Review by falmesino — APPROVED** [review]

**Review by mayun19 — APPROVED** [review]

**Review by sultanfariz — APPROVED** [review]

**Pull request merged**